src/components/dialog/createDialog.ts

```diff
@@ -0,0 +1,97 @@
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import { DNode, Widget, WidgetProperties, WidgetFactory } from '@dojo/widget-core/interfaces';
+import createWidgetBase from '@dojo/widget-core/createWidgetBase';
+import { v } from '@dojo/widget-core/d';
+
+import * as baseTheme from './styles/dialog';
+import themeableMixin, { Themeable } from '@dojo/widget-core/mixins/themeable';
+
+export interface DialogProperties extends WidgetProperties {
+	closeable?: boolean;
+	enterAnimation?: string;
+	exitAnimation?: string;
+	modal?: boolean;
+	open?: boolean;
+	title?: string;
+	underlay?: boolean;
+	onOpen?(): void;
+	onRequestClose?(): void;
+};
+
+export type Dialog = Widget<DialogProperties> & Themeable<typeof baseTheme> & {
+	onCloseClick?(): void;
+	onUnderlayClick?(): void;
+};
+
+export interface DialogFactory extends WidgetFactory<Dialog, DialogProperties> { };
+
+const createDialog: DialogFactory = createWidgetBase.mixin(themeableMixin).mixin({
+	mixin: {
+		baseTheme,
+
+		onCloseClick: function (this: Dialog) {
+			const closeable = this.properties.closeable || typeof this.properties.closeable === 'undefined';
+			closeable && this.properties.onRequestClose && this.properties.onRequestClose();
+		},
+
+		onUnderlayClick: function (this: Dialog) {
+			!this.properties.modal && this.onCloseClick && this.onCloseClick();
+		},
+
+		getChildrenNodes: function (this: Dialog): DNode[] {
+			const {
+				closeable = true,
+				enterAnimation = 'dialogShow',
```

Comment:
So I take that back, looks like they are not exported so couldn't use either of those.